modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/related/models/test/CPDefinitionSystemObjectRelatedModelsProviderTest.java

```diff
@@ -81,13 +171,93 @@ protected ObjectDefinition getSystemObjectDefinition() throws Exception {
+		throws Exception {
+
+		ObjectEntry objectEntry = ObjectEntryTestUtil.addObjectEntry(
+			groupId, _objectDefinition.getObjectDefinitionId(),
+			Collections.emptyMap());
+
+		ObjectRelationship objectRelationship = addObjectRelationship(
+			_objectDefinition, _systemObjectDefinition,
+			ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+			ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		insertIntoOrUpdateExtensionTable(
+			objectEntry.getObjectEntryId(), _cpDefinitionIds[0],
+			_systemObjectDefinition.getObjectDefinitionId());
+		insertIntoOrUpdateExtensionTable(
+			objectEntry.getObjectEntryId(), _cpDefinitionIds[1],
+			_systemObjectDefinition.getObjectDefinitionId());
+
+		ObjectRelationshipTestUtil.assertSearchRelatedModels(
+			0, objectRelatedModelsProvider,
+			objectRelationship.getObjectRelationshipId(),
+			objectEntry.getObjectEntryId(),
+			String.valueOf(RandomTestUtil.randomInt()));
+		ObjectRelationshipTestUtil.assertSearchRelatedModels(
+			1, objectRelatedModelsProvider,
+			objectRelationship.getObjectRelationshipId(),
+			objectEntry.getObjectEntryId(), "Matthew");
+		ObjectRelationshipTestUtil.assertSearchRelatedModels(
+			2, objectRelatedModelsProvider,
+			objectRelationship.getObjectRelationshipId(),
+			objectEntry.getObjectEntryId(), "Ma");
+
+		objectRelationshipLocalService.deleteObjectRelationship(
+			objectRelationship.getObjectRelationshipId());
+	}
+
+	private void _testSystemObjectEntryMtoMObjectRelatedModels(long groupId)
+		throws Exception {
+
+		ObjectEntry objectEntry = ObjectEntryTestUtil.addObjectEntry(
+			groupId, _objectDefinition.getObjectDefinitionId(),
+			Collections.emptyMap());
+
+		ObjectRelationship objectRelationship = addObjectRelationship(
+			_objectDefinition, _systemObjectDefinition,
+			ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+			ObjectRelationshipConstants.TYPE_MANY_TO_MANY);
+
+		ObjectRelationshipTestUtil.addObjectRelationshipMappingTableValues(
+			objectRelationship.getObjectRelationshipId(),
+			objectEntry.getPrimaryKey(), _cpDefinitionIds[0]);
+		ObjectRelationshipTestUtil.addObjectRelationshipMappingTableValues(
+			objectRelationship.getObjectRelationshipId(),
+			objectEntry.getPrimaryKey(), _cpDefinitionIds[1]);
+
+		ObjectRelationshipTestUtil.assertSearchRelatedModels(
+			0, objectRelatedModelsProvider,
+			objectRelationship.getObjectRelationshipId(),
+			objectEntry.getPrimaryKey(), StringUtil.randomString());
+		ObjectRelationshipTestUtil.assertSearchRelatedModels(
+			1, objectRelatedModelsProvider,
+			objectRelationship.getObjectRelationshipId(),
+			objectEntry.getPrimaryKey(), "Matthew");
+		ObjectRelationshipTestUtil.assertSearchRelatedModels(
+			2, objectRelatedModelsProvider,
+			objectRelationship.getObjectRelationshipId(),
+			objectEntry.getPrimaryKey(), "Ma");
+
+		objectRelationshipLocalService.deleteObjectRelationship(
+			objectRelationship.getObjectRelationshipId());
+	}
```

Comment:
@rijema those two methods look almost the same, please try to join them into one to avoid duplications.

Thank you!

Reply:
I merged both, and created an if/else condition to consider the X_to_X relationship it is now receiving.